progressist/__init__.py

```diff
@@ -115,12 +115,28 @@ def speed(self):
         """Number of iterations per second."""
         return Float(1.0 / self.avg if self.avg else 0)
 
+    @property
+    def throttled(self):
+        if not self.throttle:
+            return
+        if isinstance(self.throttle, (int, float)):
+            throttle = self.throttle
+            if isinstance(self.throttle, float):
+                throttle = max(1, self.total * self.throttle)
+            throttle = self._last_render + throttle
+            if self.done < throttle:
+                if not self.total or throttle <= self.total:
```

Comment:
I've tried to set `_next_render` instead, but I end up with the same subtleties: we still need to check done vs total for the edge case; plus we have two scenarios where we want to set `_next_render`: when it's not set and when we are going to render, which seems to make the code even more complex.

Also tried to return True by default, and check for cases where we need to return False: the issue is that when we return False (means "ok to render") then we also need to compute `_last_render`, so finally we want to return False only once.

So I'm going to merge as is, but if you have a better pattern in mind and energy to try it, that's still welcome :)